sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2082,6 +2082,64 @@ case class UnBase64(child: Expression)
   }
 }
 
+object Decode {
+  def createExpr(params: Seq[Expression]): Expression = {
+    params.length match {
+      case 0 => StringDecode(Literal.create(null, StringType), Literal.create(null, StringType))
+      case 1 => StringDecode(params.head, Literal.create(null, StringType))
```

Comment:
do we really allow 0 or 1 parameter before?

Reply:
Let's preserve the `AnalysisException(s"Invalid number of arguments for function decode. ")`